control-board/src/tasks/imu.rs

```diff
@@ -0,0 +1,87 @@
+use embassy_stm32::Peripheral;
+use embassy_stm32::exti::ExtiInput;
+use embassy_stm32::gpio::{Input, Pull};
+use embassy_stm32::spi::{SckPin, MisoPin, MosiPin};
+use embassy_sync::{pubsub::Publisher, blocking_mutex::raw::RawMutex};
+
+use nalgebra::Vector3;
+
+use static_cell::StaticCell;
+
+use crate::drivers::imu::bmi085::{Bmi085, GyroRange, GyroBandwidth, GyroIntMap, GyroIntPinActiveState, GyroIntPinMode};
+use crate::drivers::imu::{GyroFrame, AccelFrame};
+
+use crate::pins::{ImuSpi, ImuTxDma, ImuRxDma, ImuAccelCsPin, ImuGyroCsPin, ImuAccelIntPin, ImuGyroIntPin};
+
+#[link_section = ".axisram.buffers"]
+static IMU_BUFFER_CELL: StaticCell<[u8; 8]> = StaticCell::new();
+// static mut IMU_BUF: [u8; 8] = [0; 8];
+
+// #[embassy_executor::task]
+pub async fn imu_task<'a,
+        GyroMutex: RawMutex,
+        const GYRO_CAP: usize,
+        const GYRO_NSUB: usize,
+        const GYRO_NPUB: usize,
+        AccelMutex: RawMutex,
+        const ACCEL_CAP: usize,
+        const ACCEL_NSUB: usize,
+        const ACCEL_NPUB: usize>(
+    gyro_pub: Publisher<'a, GyroMutex, GyroFrame, GYRO_CAP, GYRO_NSUB, GYRO_NPUB>,
+    accel_pub: Publisher<'a, AccelMutex, AccelFrame, ACCEL_CAP, ACCEL_NSUB, ACCEL_NPUB>,
+    imu_spi: impl Peripheral<P = ImuSpi> + 'a,
+    sck: impl Peripheral<P = impl SckPin<ImuSpi>> + 'a,
+    mosi: impl Peripheral<P = impl MosiPin<ImuSpi>> + 'a,
+    miso: impl Peripheral<P = impl MisoPin<ImuSpi>> + 'a,
+    txdma: impl Peripheral<P = ImuTxDma> + 'a,
+    rxdma: impl Peripheral<P = ImuRxDma> + 'a,
+    accel_cs: impl Peripheral<P = ImuAccelCsPin> + 'a,
+    gyro_cs: impl Peripheral<P = ImuGyroCsPin> + 'a,
+    accel_int_pin: impl Peripheral<P = ImuAccelIntPin> + 'a,
+    gyro_int_pin: impl Peripheral<P = ImuGyroIntPin> + 'a,
+    accel_int: impl Peripheral<P = <ImuAccelIntPin as embassy_stm32::gpio::Pin>::ExtiChannel> + 'a,
+    gyro_int: impl Peripheral<P = <ImuGyroIntPin as embassy_stm32::gpio::Pin>::ExtiChannel> + 'a,
+    ) {
+    
+    let imu_buf = IMU_BUFFER_CELL.init([0; 8]);
+    let mut imu = Bmi085::new_from_pins(imu_spi, sck, mosi, miso, txdma, rxdma, accel_cs, gyro_cs, imu_buf);
+
```

Comment:
Should we add the self_test here? Related but accel needs to have a dummy read to initialize to SPI mode and self_test does that

Reply:
created an init function to be called. I don't like having new() be async because that breaks some meta patterns in Rust that are seemingly unofficial, but still.